### PR TITLE
bugfix/21524-column-ordinal-panning

### LINKED ISSUE #21524: Drag does not work correctly
#### Expected behaviour

There are problems when using a bar graph. The problem is that when dragging, there is no way to scroll to the very beginning.
Demo:
https://www.highcharts.com/demo/stock/column

watch the video
https://github.com/highcharts/highcharts/assets/11316123/052af76f-00d4-429a-88a9-2b0534053d37

In the demo I try to scroll through drag to the very beginning, but it doesn’t work.

There is also a problem with "xAxix.ordinal.positions", when dragging the value "undefined" may appear

#### Actual behaviour
will not scroll to the beginning when dragging

#### Live demo with steps to reproduce
https://www.highcharts.com/demo/stock/column
https://github.com/highcharts/highcharts/assets/11316123/052af76f-00d4-429a-88a9-2b0534053d37

#### Product version
11.x.x

#### Affected browser(s)
all

### PR DESCRIPTION
**Pull request by bm64**
Fixed #21524, in some cases panning to axis minimum was impossible in data grouped ordinal column chart.

### CODE REVIEW

**Review by kamil-musialowski — APPROVED** [review]
Looking good!

**Review by pawellysy — APPROVED** [review]

File: samples/unit-tests/chart/panning/demo.js [hunk 1]
```diff
@@ -726,3 +726,43 @@ QUnit.test(
         assert.strictEqual(actualMin, -75, 'Min must be -75; not panning');
     }
 );
+
+QUnit.test('Column chart panning.', function (assert) {
```
Comment (pawelfus) [inline samples/unit-tests/chart/panning/demo.js:730]: Could you move the test to `Ordinal axis panning`? It seems to be almost exactly the same with some minor changes.

**Review by pawelfus — CHANGES_REQUESTED** [review]
Thanks! Just some minor errands, see below

**Review by DJTechnoo — APPROVED** [review]
looks great!

File: samples/unit-tests/chart/panning/demo.js [hunk 2]
```diff
@@ -356,66 +356,49 @@ QUnit.test('Stock panning (#6276, #21319)', function (assert) {
...
             chart.xAxis[0].min,
-            'Chart should pan horizontally.'
+            'Chart should pan horizontally, when data is equally spaced #13334.'
+        );
+
+        assert.equal(
+            chart.xAxis[0].min,
+            0,
+            `It should be possible to pan to the axis minimum in a data grouped
+            ordinal column chart, #21524.`
```
Comment (pawelfus) [inline samples/unit-tests/chart/panning/demo.js:401]: When testing grouped data, please set `forced=true`. And keep the old test. What I meant in the previous review is: to move new test to the old test, so we will reuse the chart (it takes most of the time in our tests). Something like this:

```ts
const chart = new chart(..);
controller.pan(..);
assert.notEqual(..);

chart.series[0].update({ data, dataGrouping });
controller.pan(..);
assert.equal(..);
```

**Review by pawelfus — CHANGES_REQUESTED** [review]
See my comment below

**Review by pawelfus — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**